### ISSUE TITLE
🐛 Disabled checklist story looks identical to default

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#11422]
## 🐛 Bug

The disabled checklist story looks identical to the default checklist story.

## 🦋 Expected Behaviour

The disabled checklist should look different - kind of grey I guess?

## 📋 Steps to Reproduce

1. https://www.chromatic.com/component?appId=61e099a1bb1465003a73d3ce&csfId=forms-src-components-checklist--disabled-checklist&buildNumber=8605&k=66d0c879e23fad5c3ddaecaf-1200px-interactive-true&h=4&b=-2

## 📸 Screenshot

![image](https://github.com/user-attachments/assets/f099c1db-9aee-4061-b401-3ef2c014edd6)

## ✅ Acceptance Criteria

- [ ] Disabled checklist looks different from default

**Comment by petertgiles** [comment]
@JoshBeveridge Could you please look at this and suggest a design change?

**Comment by JoshBeveridge** [comment]
@petertgiles couple things to note here:
* We're limited in the scope of grays we can use due to contrast requirements, which led us to use this darker shade, but I agree that it's overly similar
* As far as I remember, we're not using disabled states for anything because it's bad for a11y, however Jerbo and I are working on a middle-ground solution where we show a placeholder element in place of the input that provides extra context:
![image](https://github.com/user-attachments/assets/d2172ec4-efcd-4d4e-9bec-80198e375ee7)

That's not necessarily final, as we still need to have further discussions about it, but feel free to provide feedback.

**Comment by gobyrne** [comment]
@JoshBeveridge I assume work on this will get rolled into component changes? Can you close this issue if you agree?

**Comment by mnigh** [comment]
> > If we are not using it anywhere (and should not for a11y reasons), can we close this issue and create an issue to remove the storybook story?
> 
> Yes, [form fields really should never be disabled](https://adrianroselli.com/2024/02/dont-disable-form-controls.html#Fields). There is almost always a better solution which is typically, good error messages and validation.

Instances where the `Checklist` component has the `disabled` prop:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/169c1918146a167bfd59671a39723074d315d971/apps/web/src/pages/CommunityInterests/sections/FindANewCommunity.tsx#L283-L299

https://github.com/GCTC-NTGC/gc-digital-talent/blob/169c1918146a167bfd59671a39723074d315d971/apps/web/src/pages/EmployeeProfile/components/CareerObjective/CareerObjectiveSection.tsx#L680-L688

https://github.com/GCTC-NTGC/gc-digital-talent/blob/169c1918146a167bfd59671a39723074d315d971/apps/web/src/pages/EmployeeProfile/components/NextRoleSection/NextRoleSection.tsx#L676-L683

**Comment by gobyrne** [comment]
> @petertgiles If we are not using it anywhere (and should not for a11y reasons), can we close this issue and create an issue to remove the storybook story?

I don't think so. We are reserving a place in our component library for "visually disabled" variants. These would be applied to inputs that behave as Eric recommends: still functions with good error messages and validation. The "visually disabled" colors just signals that you are wasting your time by interacting with it. You can still interact with the component, you'll just get error messages (or similar) when you do. This is similar to this section on the dashboard.
<img width="752" height="416" alt="Image" src="https://github.com/user-attachments/assets/47214b0a-60f7-40d1-a16e-0502a8f575bb" />

 I think this story was reasonably brought forward by Peter to flag that the visual diff was effectively non-existant. Design has a plan, but we haven't needed to clarify it since these are effectively not used yet.

Now, the 3 instances that Matt found appear to be places where there are no options to pick (for workstream) until you select the community. I think this means the checkbox group wouldn't even be an interactable element on the page. Either hiding these entirely until community is selected or using a placeholder like Josh proposes in his comment both seem reasonable to me. But to be clear, we should not be setting the html input to disabled, for the reasons Eric suggests. Greying-out the component until it is interactable, or until it's possible to set the input value to something that will not throw an error would meet our planned intent of the "visually disabled" variant.

**Comment by mnigh** [comment]
Not just in stories and for checklist, but also in production fields (Level number field in classification form).

## Screenshot
<img width="1655" height="878" alt="Image" src="https://github.com/user-attachments/assets/dceecc2e-9754-4184-b1c4-84b2ba8aa375" />

**Comment by gobyrne** [comment]
Yeah, this follows the same pattern as the workstreams: there's no valid option until you pick the "Group". We shouldn't be using disabled though.

I'd be inclined to invest in a new classification picker widget. It's something we'll need for our expanded whole-of-gov coverage anyway. Doesn't help the workstream stuff though.
